inc/alert.class.php

```diff
@@ -210,79 +210,126 @@ public static function findAllToNotify($params = []) {
          foreach ($fndgroup_user as $group) {
             $fndgroup[] = $group['groups_id'];
          }
-         $fndgroup = implode(',', $fndgroup);
       }
       if (empty($fndgroup)) {
-         $fndgroup = "-1";
+         $fndgroup = [-1];
       }
 
       // filters for query
-      $targets_sql           = "";
-      $login_sql             = "";
-      $login_show_hidden_sql = " `$utable`.`id` IS NULL ";
-      $entity_sql            = "";
-      $show_helpdesk_sql     = '';
-      $show_central_sql      = '';
+      $targets_sql           = [];
+      $login_sql             = [];
+      $login_show_hidden_sql = ["{$utable}.id" => null];
+      $entity_sql            = [];
+      $show_helpdesk_sql     = [];
+      $show_central_sql      = [];
       if (isset($_SESSION['glpiID']) && isset($_SESSION['glpiactiveprofile']['id'])) {
-         $targets_sql = "AND (
-                           `$ttable`.`itemtype` = 'Profile'
-                           AND (
-                              `$ttable`.`items_id` = ".$_SESSION['glpiactiveprofile']['id']."
-                              OR `$ttable`.`all_items` = 1
-                           )
-                           OR `$ttable`.`itemtype` = 'Group'
-                              AND `$ttable`.`items_id` IN ($fndgroup)
-                           OR `$ttable`.`itemtype` = 'User'
-                              AND `$ttable`.`items_id` = ".$_SESSION['glpiID']."
-                        )";
+         $targets_sql = [
+            'OR' => [
+               [
+                  'AND' => [
+                     "$ttable.itemtype" => 'Profile',
+                     "$ttable.items_id" => $_SESSION['glpiactiveprofile']['id'],
```

Comment:
Condition on `all_items` is missing.
```suggestion
                     'OR' => [
                         "$ttable.items_id" => $_SESSION['glpiactiveprofile']['id'],
                         "$ttable.all_items" => 1,
                     ],
```